### ISSUE TITLE
False positives for declaration-empty-line-before

### ISSUE THREAD

**Issue by hornta** [stylelint/stylelint#4735]
> Clearly describe the bug

There are false positives for when using less mixins together with normal css properties.

> Which rule, if any, is the bug related to?

declaration-empty-line-before

> What code is needed to reproduce the bug?

```css
.foo{
  .my-mixing();
  color: white;
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "declaration-empty-line-before": "always"
  }
}
```

> Which version of stylelint are you using?

13.3.3

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `npx stylelint "**/*.less"`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to CSS mixins.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
test.css
3:3    Expected empty line before declaration
```

**Comment by jeddy3** [comment]
@hornta Thanks for the report and for using the template.

The rule is behaving as expected. With `"always"` there should be an empty line before the declaration and there is not.

You can either use the `"never"` option, or create a [plugin](https://stylelint.io/developer-guide/plugins) (similar to the ones in [stylelint-scss](https://github.com/kristerkari/stylelint-scss)) that understands Less mixins so that you can continue to use `"always"` like so:

```json
{
  "rules": {
    "less/declaration-empty-line-before": ["always", { "except": ["after-mixin"] }]
  }
}
```

**Comment by hornta** [comment]
Hi. I believe I had the incorrect stylelint config. It should be 
```
"declaration-empty-line-before": [
      "always",
      {
        except: ["after-declaration", "first-nested"],
        ignore: ["after-comment", "inside-single-line-block"],
      },
    ]
```
as in the stylelint-config-standard. I will update the issue with this.

**Comment by hornta** [comment]
This had something to do with me using the incorrect vscode stylelint packade since there are too many out there. I updated to the official one the error is gone.

**Comment by jeddy3** [comment]
> since there are too many out there. 

Yes, that's an issue. There an initiative (https://github.com/stylelint/vscode-stylelint/issues/31) to resolve this.
